### PR TITLE
Tx search levels

### PR DESCRIPTION
**Pull request by NaderMahdi**
* *Added tx search levels**

    enc_mode > ENC_M1
    tx_search_level = TX_SEARCH_ENC_DEC;
    Otherwise
    tx_search_level = TX_SEARCH_FULL_LOOP;

    // Tx_search Level                       Settings
    // 0                                              OFF
    // 1                                              Tx search at encdec
    // 2                                              Tx search at inter-depth
    // 3                                              Tx search at full loop

    // Set tx search skip weights (MAX_MODE_COST: no skipping; 0: always skipping)
     -  tx_weight = MAX_MODE_COST;

    // Set tx search reduced set falg (0: full tx set; 1: reduced tx set)
    enc_mode = ENC_M1
       - tx_search_reduced_set = 1;
    Otherwise
      -  tx_search_reduced_set = 0;

### CODE REVIEW

File: Source/Lib/Codec/EbProductCodingLoop.c [hunk 1]
```diff
@@ -2441,6 +2471,312 @@ EbBool allowed_ns_cu(
...
+        memset(count_non_zero_coeffs[2], 0, sizeof(uint32_t)*MAX_NUM_OF_TU_PER_CU);
+
+
+        ProductFullLoopTxSearch(
+            candidateBuffer,
+            context_ptr,
+            picture_control_set_ptr
+        );
+
+
```
Comment (andrey-norkin) [inline Source/Lib/Codec/EbProductCodingLoop.c]: Somewhat sparse code

Comment (NaderMahdi) [reply Source/Lib/Codec/EbProductCodingLoop.c]: Added init_candidate_buffer() function to reduce the sparse code.

**Comment by NaderMahdi** [comment]
@andrey-norkin - Any other objection to merge this PR?

**Comment by hassount** [comment]
lgtm

**Pull request merged**